content/java/concepts/compiler/compiler.md

```diff
@@ -0,0 +1,37 @@
+---
+Title: 'Compiler'
+Description: 'Java compilers typically take Java source code and produce class files containing platform-neutral Java bytecode that can be executed by the Java Virtual Machine (JVM).'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Command Line'
+  - 'Developer Tools'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Java **compilers** are programs that take Java source code and produce class files containing platform-neutral Java bytecode that can be executed by the Java Virtual Machine (JVM).
+
+The JVM does not interpret source code. Instead, it interprets what is called Java bytecode (a language somewhere between human-readable source code, and [machine code](https://www.codecademy.com/resources/docs/general/machine-code)), which is specific to a particular computer. Unlike machine code, bytecode is platform-neutral meaning the same bytecode can be interpreted by any JVM running on any sort of computer system. This is what makes compiled Java programs portable.
+
+Most of these compilers do little to no optimization of the code, leaving that task for the JVM to do at run time. The JVM loads the bytecode and either interprets it, or just-in-time (JIT) compiles it to machine code, and then possibly optimizes it.
+
+## Compiling a Java File
+
+The Java compiler is executed on the command line in [Unix](https://www.codecademy.com/resources/docs/general/unix) or DOS shells as follows:
+
+```shell
+javac ProgramName.java
```

Comment:
```suggestion
javac ProgramName.java

Java options ProgramName.java
```

Reply:
referred to: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javac.html

Since an option is mentioned below, thought it would be useful to include in this shell code block